cartoframes/context.py

```diff
@@ -404,6 +405,28 @@ def fetch(self, query, decode_geom=False):
               `Shapely <https://github.com/Toblerity/Shapely>`__
               object that can be used, for example, in `GeoPandas
               <http://geopandas.org/>`__.
+            query_columns (list, optional): A list of `cartoframes.column.Column` to infer data types. You should
+              create `Column` instances this way: Column('column_name', pgtype='the_database_type') where
+              `the_database_type` is mapped using this dictionary:
```

Comment:
I'm a little tempted to make `Column` more accessible (like `cartoframes.Column`), but in general people aren't expected to be using these too much so I think it makes sense to have them a little more buried lie this.

Reply:
Actually, a better form would be to copy what pandas does in [`pd.read_csv`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html):

> dtype : Type name or dict of column -> type, optional
> Data type for data or columns. E.g. {‘a’: np.float64, ‘b’: np.int32, ‘c’: ‘Int64’} Use str or object together with suitable na_values settings to preserve and not interpret dtype. If converters are specified, they will be applied INSTEAD of dtype conversion.

This allows the user to accept whatever is inferred for most types, but specify a few that don't convert as intended.

And the behavior could be if a column is specified that isn't in the query, a warning is issued that that column wasn't present in the query.